cmd/snap/error.go

```diff
@@ -190,3 +220,149 @@ If you understand and want to proceed repeat the command including --classic.
 
 	return msg, nil
 }
+
+func snapRevisionNotAvailableMessage(kind, snapName, action, arch, channel string, releases []interface{}) string {
+	// releases contains all available (arch x channel)
+	// as reported by the store through the daemon
+	req, err := snap.ParseChannel(channel, arch)
+	if err != nil {
+		msg := fmt.Sprintf(i18n.G("requested an invalid channel for snap %[1]q.\nPlease use 'snap info %[1]s' to list available releases."), snapName)
```

Comment:
Messages here might be tuned to be similar to the suggestions above.

It's curious that we have this duplication, though. The other messages seem to state facts very similar to the ones stated here. I wonder if there's a chance to unify this logic too so that we have a single path for all of these cases and a single message for each case.

Reply:
the code before is for the (old) fallback kind (or when the user gave --revision), here we deal with the new kinds, a couple messages look similar but here generally we have more information or can make more assumptions so the messages are bit more verbose, can always give a little bit more detail.

Reply:
I changed this one to follow the other style, it's a corner case, in which the user specified something that is not valid even syntactically as a channel